test/go/src/tests/api-v200_test.go

```diff
@@ -0,0 +1,258 @@
+						for j := 0; j < len(testsToRunContent.Tests[i].Files); j++ {
+							// Read the snippet
+							data, err := ioutil.ReadFile(filepath.Join(testDir, testsToRunContent.Tests[i].Files[j]))
+							if err != nil {
+								t.Errorf("FAIL: failed reading %s: %v", filepath.Join(testDir, testsToRunContent.Tests[i].Files[j]), err)
+								testYamlComplete = false
+								continue
+							}
+							if j > 0 {
+								// Ensure appropriate line breaks
+								f.WriteString("\n")
+							}
+
+							// Add snippet to yaml file
+							f.Write(data)
+						}
+
+						if !testYamlComplete {
+							f.Close()
+							continue
+						}
+
+						// Read the created yaml file, ready for converison to json
+						data, err := ioutil.ReadFile(filepath.Join(testTempDir, testsToRunContent.Tests[i].FileName))
+						if err != nil {
+							t.Errorf("  FAIL: unable to read %s: %v", testsToRunContent.Tests[i].FileName, err)
+							f.Close()
+							continue
+						}
+
+						f.Close()
+
+						// Convert the yaml file to json
+						yamldoc, err := yaml.YAMLToJSON(data)
+						if err != nil {
+							t.Errorf("  FAIL : %s : failed to convert to json : %v", testsToRunContent.Tests[i].FileName, err)
+							continue
+						}
+
+						// validate the test yaml against the schema
+						if err = schema.Validate(bytes.NewReader(yamldoc)); err != nil {
+							if testsToRunContent.Tests[i].ExpectOutcome == "PASS" {
+								t.Errorf("  FAIL : %s : %s : Validate failure : %s", testsToRunContent.Tests[i].FileName, testJsonContent.SchemaFile, err)
+							} else if testsToRunContent.Tests[i].ExpectOutcome == "" {
+								t.Errorf("  FAIL : %s : No expected ouctome was set : %s  got : %s", testsToRunContent.Tests[i].FileName, testsToRunContent.Tests[i].ExpectOutcome, err.Error())
+							} else if !strings.Contains(err.Error(), testsToRunContent.Tests[i].ExpectOutcome) {
+								t.Errorf("  FAIL : %s : %s : Did not fail as expected : %s  got : %s", testsToRunContent.Tests[i].FileName, testJsonContent.SchemaFile, testsToRunContent.Tests[i].ExpectOutcome, err.Error())
+							} else {
+								passTests++
+								t.Logf("PASS : %s : %s: %s", testsToRunContent.Tests[i].FileName, testJsonContent.SchemaFile, testsToRunContent.Tests[i].ExpectOutcome)
+							}
+						} else if testsToRunContent.Tests[i].ExpectOutcome == "" {
+							t.Errorf("  FAIL : %s : devfile was valid - No expected ouctome was set.", testsToRunContent.Tests[i].FileName)
+						} else if testsToRunContent.Tests[i].ExpectOutcome != "PASS" {
+							t.Errorf("  FAIL : %s : devfile was valid - Expected Error not found :  %s", testsToRunContent.Tests[i].FileName, testsToRunContent.Tests[i].ExpectOutcome)
+						} else {
+							passTests++
+							t.Logf("PASS : %s : %s", testsToRunContent.Tests[i].FileName, testJsonContent.SchemaFile)
+						}
+						f.Close()
+					}
+				}
+			}
+			t.Logf("%s : %d of %d tests passed", testJsonFile.Name(), passTests, totalTests)
```

Comment:
It's also useful to see number of skipped tests

Reply:
Sure. will also address this is a subsequent pull request.